File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int ix = 0; ix != nums.length; ++ix) {
            if (map.containsKey(nums[ix])) {
                return new int[] { ix, map.get(nums[ix])
                };
            }
            map.put(target - nums[ix], ix);
        }
        return new int[] { -1, -1 };
    }
}
// @lc code=end

File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start

#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (auto ix = 0; ix != nums.size(); ++ix) {
            auto iter = seen.find(target - nums.at(ix));
            if (iter != seen.end()) {
                return {ix, iter->second};
            }
            seen.insert(make_pair(nums.at(ix), ix));
        }
        return {-1, -1};
    }
};
// @lc code=end
